File: Editor/src/main/java/org/turbomedia/turboedit/editor/panes/timeline/ClipContainer.java
package org.turbomedia.turboedit.editor.panes.timeline;

import javafx.geometry.Insets;
import javafx.scene.layout.*;
import javafx.scene.paint.Color;
import org.turbomedia.turboedit.editor.misc.StyleManager;

import java.io.IOException;

public class ClipContainer extends HBox {

    public ClipContainer(LayerBox.LayerType type) {
        StyleManager.RegisterEvent((theme) -> {
            Color color = switch (theme) {
                case DARK -> Color.color(1, 1, 1, 0.05);
                case LIGHT -> Color.color(0, 0, 0, 0.05);
            };

            var background = new Background(new BackgroundFill(color, new CornerRadii(10), Insets.EMPTY));

            setBackground(background);
        });

        setSpacing(4);
        setMaxWidth(Double.MAX_VALUE);
        HBox.setHgrow(this, Priority.ALWAYS);

        StyleManager.RegisterEvent((theme) -> {
            getChildren().clear();

            try {
                getChildren().add(new ClipCard(type));
            } catch (IOException e) {
                throw new RuntimeException(e);
            }
        });
    }
}

File: Editor/src/main/kotlin/turboedit/editor/panes/timeline/ClipContainer.kt
package turboedit.editor.panes.timeline

import javafx.geometry.Insets
import javafx.scene.layout.*
import javafx.scene.paint.Color
import javafx.scene.paint.Paint
import turboedit.editor.events.EventSystem.EventAction
import turboedit.editor.events.EventSystem.registerListener
import turboedit.editor.events.EventType
import turboedit.editor.events.ThemeChangedEventData
import turboedit.editor.misc.StyleManager
import turboedit.editor.misc.StyleManager.callEvent
import java.io.IOException

class ClipContainer(type: LayerBox.LayerType) : HBox() {
    init {
        spacing = 4.0
        maxWidth = Double.Companion.MAX_VALUE
        setHgrow(this, Priority.ALWAYS)

        registerListener(EventType.THEME_CHANGED, EventAction { dat: Any? ->
            val data = dat as ThemeChangedEventData?
            val color = when (data!!.theme) {
                StyleManager.Theme.DARK -> Color.color(1.0, 1.0, 1.0, 0.05)
                StyleManager.Theme.LIGHT -> Color.color(0.0, 0.0, 0.0, 0.05)
                else -> {}
            }

            val background = Background(BackgroundFill(color as Paint?, CornerRadii(10.0), Insets.EMPTY))

            setBackground(background)
            children.clear()

            try {
                children.add(ClipCard(type))
            } catch (e: IOException) {
                throw RuntimeException(e)
            }
        })

        callEvent()
    }
}
